packages/controls/src/widget_dom_events.ts

```diff
@@ -0,0 +1,275 @@
+            current_model.set('_view_count', 0)
+        }
+        console.log('Adding backbone listener for changes in view count')
+        this.listenTo(current_model, 'change:_view_count', this.update_listeners)
+        this.update_listeners()
+    }
+
+    update_listeners() {
+        // Remove all existing DOM event listeners
+        this.remove_listeners()
+        // Add watchers to any existing views of the model
+        this.attach_listeners()
+    }
+
+    remove_listeners() {
+        // Remove all of the event listeners stored in the cache.
+        for (let listener of this.get('_attached_listeners')) {
+            console.log('Removing ', listener.event, ' from ', listener.view, 'function', listener.func)
+            listener.view.el.removeEventListener(listener.event, listener.func)
+        }
+        this.set('_attached_listeners', [])
+    }
+
+    _add_listeners_to_view(view) {
+        for (let event of this.get('watched_events')) {
+            switch (this.key_or_mouse(event)) {
+                case "keyboard":
+                    this._add_key_listener(event, view)
+                    break
+                case "mouse":
+                    let handler = this._dom_click.bind(this, view)
+                    view.el.addEventListener(event, handler)
+                    // Keep track of the listeners we are attaching so that we can
+                    // remove them if needed.
+                    this._cache_listeners(event, view, handler)
+                    break
+                default:
+                    console.log('Not familiar with that message source')
+                    break
+            }
+        }
+    }
+
+    attach_listeners() {
+        // console.log("Attaching listeners")
+        let current_source = this.get('source')
+        _.each(current_source.views, (view_promise) => {
+            Promise.resolve(view_promise).then((view) => {
+                this._add_listeners_to_view(view)
+            })
+        })
+    }
+
+    _add_key_listener(event_type, view) {
+        // Key listeners should:
+        //     + Only fire when the mouse is over the element.
+        //     + Not propagate up to the notebook (because imagine you
+        //       press 'x' on your widget and cut the cell...not what
+        //       you probably want the user to experience)
+        //
+        // The approach here is to add a key listener on mouseenter and remove
+        // it on mouseleave.
+        let key_handler = (event) => {
+            // console.log('Key presses FTW!', event)
+            if (this.get('ignore_modifier_key_events') &&
+               _.contains(this.get('_modifier_keys'), event.key)) {
+                // If the key event is supposed to be ignored, then skip it.
+                return
+            }
+            this._send_dom_event(event)
+            // Need this (and useCapture in the listener) to prevent the keypress
+            // from propagating to the notebook.
```

Comment:
If we add a `event.preventDefault()` here it prevents the quote keys, slash key, home, end , page up and page down from being processed by the browser.

Reply:
Are these keypresses that you would prefer be passed on to the browser while over the widget?

Reply:
No, they are currently being processed by the widget (which is listening for them) and then also affecting the browser.  If the widget is listening for them they should not be also processed by the browser, IMHO.  For example, if you implement a scrolling function bound to the `home` key then not only does it scroll your viewer (in the widget) but also the browser will scroll the whole notebook back to the top of the notebook.

Reply:
Ah, I'll try to get that fixed tomorrow. Any idea which key(s) one of those things is bound to on a mac keyboard?

Reply:
After your recent updates, the remaining problem keys seem to be forward slash, single quote (these two trigger the browser quick find box), and pageup/pagedown/home/end (which scroll the browser page).  I'm testing with Chrome on Linux.

Seems like if you just test in the right spot in the key handler for `prevent_default_action` = `True` and do an `event.preventDefault()` it should fix this.  I had all the keys working with a modified version of `widget_dom_events.ts` in which I invoked that in the key handler.